src/error.rs

```diff
@@ -64,10 +65,17 @@ impl fmt::Display for Error {
             Error::UndefinedLineNumber {
                 src_line_number,
                 line_number,
+                ref statement_source,
             } => write!(
                 f,
-                "{}: error: non-existing line number \n GOTO {}\n      ^\n",
-                src_line_number, line_number
+                "{}: error: non-existing line number \n {}\n{:cursor$}^\n",
+                src_line_number,
+                statement_source,
+                "",
+                cursor = statement_source
+                    .find(&format!("{}", line_number))
+                    .unwrap_or(0)
+                    + 1
```

Comment:
I'd factor this out. This logic is a little more evolved. What is the cursor here?

Reply:
Did it, also commented what I am actually doing here.